From: Yves-Alexis Perez <corsac@debian.org>
Date: Fri, 16 Feb 2018 17:42:28 +0100
Subject: powerpc: add missing includes for rfi_* macros

Backported patches for PTI support on powerpc in 4.9 lead to build failure with following output:

 linux-4.9.81/arch/powerpc/kernel/entry_64.S: Assembler messages:
 linux-4.9.81/arch/powerpc/kernel/entry_64.S:260: Error: unrecognized opcode: `rfi_to_user'
 linux-4.9.81/arch/powerpc/kernel/entry_64.S:270: Error: unrecognized opcode: `rfi_to_kernel'
 linux-4.9.81/arch/powerpc/kernel/entry_64.S:885: Error: unrecognized opcode: `rfi_to_user'
 linux-4.9.81/arch/powerpc/kernel/entry_64.S:900: Error: unrecognized opcode: `rfi_to_kernel'
 linux-4.9.81/scripts/Makefile.build:398: recipe for target 'arch/powerpc/kernel/entry_64.o' failed

Upstream commit 222f20f140623ef6033491d0103ee0875fe87d35 seems to be missing
from the 4.9 backport, but it doesn't apply proplery. Until this is properly
fixed (in 4.9.82 or 4.9.83), just add the missing include.

Signed-off-by: Yves-Alexis Perez <corsac@debian.org>
---

diff --git a/arch/powerpc/kernel/entry_64.S b/arch/powerpc/kernel/entry_64.S
index 3320bcac7192..e68faa4d1b13 100644
--- a/arch/powerpc/kernel/entry_64.S
+++ b/arch/powerpc/kernel/entry_64.S
@@ -37,6 +37,11 @@
 #include <asm/tm.h>
 #include <asm/ppc-opcode.h>
 #include <asm/export.h>
+#ifdef CONFIG_PPC_BOOK3S
+#include <asm/exception-64s.h>
+#else
+#include <asm/exception-64e.h>
+#endif
 
 /*
  * System calls.

